From ff8e588a2895f4d46147c20a86b65cb9f25a3745 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Mon, 13 Oct 2014 12:28:51 +0200
Subject: [PATCH 17/25] common: change the User-Agent string to read Iridium

Continue providing Chrome/* for possible compatibility checks by the browser
and/or websites, since we really are still Chromium.
---
 chrome/common/chrome_version_info.cc | 2 +-
 content/common/user_agent.cc         | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/chrome/common/chrome_version_info.cc b/chrome/common/chrome_version_info.cc
index e197bd0..df1c1ba 100644
--- a/chrome/common/chrome_version_info.cc
+++ b/chrome/common/chrome_version_info.cc
@@ -18,7 +18,7 @@ namespace chrome {
 std::string VersionInfo::ProductNameAndVersionForUserAgent() const {
   if (!is_valid())
     return std::string();
-  return "Chrome/" + Version();
+  return "Iridium/" + Version();
 }
 
 #if defined(OS_WIN) || defined(OS_MACOSX)
diff --git a/content/common/user_agent.cc b/content/common/user_agent.cc
index 1176c48..38f2d9f 100644
--- a/content/common/user_agent.cc
+++ b/content/common/user_agent.cc
@@ -191,6 +191,10 @@ std::string BuildUserAgentFromOSAndProduct(const std::string& os_info,
       product.c_str(),
       WEBKIT_VERSION_MAJOR,
       WEBKIT_VERSION_MINOR);
+  if (strncmp(product.c_str(), "Iridium/", 8) == 0) {
+  	const char *p = strchr(product.c_str(), '/') + 1;
+  	base::StringAppendF(&user_agent, " Chrome/%s", p);
+  }
   return user_agent;
 }
 
-- 
2.4.0

